src/IREquality.cpp

```diff
@@ -106,6 +106,22 @@ IRComparer::CmpResult IRComparer::compare_scalar(T a, T b) {
         return result;
     }
 
+    if constexpr (std::is_floating_point_v<T>) {
+        // NaNs are equal to each other and less than non-nans
```

Comment:
This makes me very nervous. In IEEE754, NaN != everything, including NaN.

Reply:
This is just for matching the identity of IR nodes... in particular the previous revision considered NaN **is** equal to everything

Reply:
We could discuss --- in a future PR, perhaps --- extending the set of results for `IRComparer` to include `NotEqual`, which would allow it to express partial orders (like the one that exists in IEEE 754). I have no idea how far-reaching that change would be.

Reply:
For the purposes of this function we could also just bit-cast the float to a uint64 and compare the bits. The important thing here is that things that are not equal compare as not equal.